### PR TITLE
Update Circle CI

### PR DESCRIPTION
**Pull request by jfrerich**
#### Summary
Sync to latest https://github.com/mattermost/mattermost-plugin-demo/blob/master/.circleci/config.yml

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,27 +1,52 @@
...
+          filters:
+            tags:
+              only: /^v.*/
+            branches:
+              ignore: /.*/
+          context: plugin-ci
+          requires:
+            - plugin-ci/lint
+            - plugin-ci/coverage
+            - plugin-ci/build
```
Comment (hanzei) [inline .circleci/config.yml]: As mentioned in https://community.mattermost.com/core/pl/31bk781yxib1ijgsayrhhsgrnw I don't think this is needed any longer. Would you mind waiting to merge this PR and let me confirm?

Comment (hanzei) [reply .circleci/config.yml]: This job should be removed: https://community.mattermost.com/core/pl/h917appqmpgxxkr5k4mor9f7py

Comment (levb) [reply .circleci/config.yml]: @jfrerich ^^

Comment (jfrerich) [reply .circleci/config.yml]: @hanzei, @levb - removal of `deploy-release` has been commited.

**Comment by levb** [comment]
@jfrerich @marianunez I am a little confused as to why the build is still failing, thought the intention of this PR would be to have the build green. Or are there separate PRs for the build fixes?

**Comment by jfrerich** [comment]
@levb, the PR is pointing to master which doesn't have the changes from https://github.com/mattermost/mattermost-plugin-agenda/pull/2.  I can change the reference branch to https://github.com/mattermost/mattermost-plugin-agenda/tree/pluginReview.  That PR will have a clean build.

**Pull request merged**